c_models/src/my_models/implementations/my_neuron_impl_semd.h

```diff
@@ -128,69 +133,78 @@ static bool neuron_impl_do_timestep_update(index_t neuron_index,
-            input_type->my_multiplicator[i] = 0;
+    bool spike = false;
+    for (uint32_t i = n_steps_per_timestep; i > 0; i--) {
+
+        // Get the voltage
+        state_t voltage = neuron_model_get_membrane_voltage(neuron);
+
+        // Get the exc and inh values from the synapses
+        input_t* exc_input_values =
+                synapse_types_get_excitatory_input(synapse_type);
+        input_t* inh_input_values =
+                synapse_types_get_inhibitory_input(synapse_type);
+
+        // Set the inhibitory my_multiplicator value
+        for (int i = 0; i < NUM_INHIBITORY_RECEPTORS; i++) {
+            if ((inh_input_values[i] >= 0.01) &&
+                    (input_type->my_multiplicator[i] == 0) &&
+                    (input_type->my_inh_input_previous[i] == 0)) {
+                input_type->my_multiplicator[i] = exc_input_values[i];
+            } else if (inh_input_values[i] < 0.01) {
+                input_type->my_multiplicator[i] = 0;
+            }
+            input_type->my_inh_input_previous[i] = inh_input_values[i];
         }
-        input_type->my_inh_input_previous[i] = inh_input_values[i];
-    }
 
-    // Sum g_syn contributions from all receptors for recording
-    REAL total_exc = 0;
-    REAL total_inh = 0;
+        // Sum g_syn contributions from all receptors for recording
+        REAL total_exc = 0;
+        REAL total_inh = 0;
 
-    for (int i = 0; i < NUM_EXCITATORY_RECEPTORS; i++) {
-        total_exc += exc_input_values[i];
-    }
-    for (int i = 0; i < NUM_INHIBITORY_RECEPTORS; i++) {
-        total_inh += inh_input_values[i];
-    }
+        for (int i = 0; i < NUM_EXCITATORY_RECEPTORS; i++) {
+            total_exc += exc_input_values[i];
+        }
+        for (int i = 0; i < NUM_INHIBITORY_RECEPTORS; i++) {
+            total_inh += inh_input_values[i];
+        }
+
+        // Do recording if on first step
+        if (i == n_steps_per_timestep) {
+            neuron_recording_record_accum(V_RECORDING_INDEX, neuron_index, voltage);
+            neuron_recording_record_accum(GSYN_EXC_RECORDING_INDEX, neuron_index, total_exc);
+            neuron_recording_record_accum(GSYN_INH_RECORDING_INDEX, neuron_index, total_inh);
+        }
+
+        // This changes inhibitory to excitatory input
+        for (int i = 0; i < NUM_INHIBITORY_RECEPTORS; i++) {
+            inh_input_values[i] = -inh_input_values[i] * SCALING_FACTOR
+                    * input_type->my_multiplicator[i];
+        }
 
-    // Call functions to get the input values to be recorded
-    neuron_recording_record_accum(GSYN_EXC_RECORDING_INDEX, neuron_index, total_exc);
-    neuron_recording_record_accum(GSYN_INH_RECORDING_INDEX, neuron_index, total_inh);
+        // update neuron parameters
+        state_t result = neuron_model_state_update(
+                NUM_EXCITATORY_RECEPTORS, exc_input_values,
+                NUM_INHIBITORY_RECEPTORS, inh_input_values, external_bias, neuron);
```

Comment:
Same comment as in sPyNNaker branch, would it be sensible to pass the loop variable into here by default?

Reply:
See https://github.com/SpiNNakerManchester/sPyNNaker/pull/817#discussion_r417183511